pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -321,15 +323,180 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
 		// Pods may fail to schedule because of volumes conflicting with other pods on same node.
 		// Once running pods are deleted and volumes have been released, the unschedulable pod will be schedulable.
 		// Due to immutable fields `spec.volumes`, pod update events are ignored.
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}, QueueingHintFn: pl.isSchedulableAfterPodDeleted},
 		// A new Node may make a pod schedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add}},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add}, QueueingHintFn: pl.isSchedulableAfterNodeAdded},
 		// Pods may fail to schedule because the PVC it uses has not yet been created.
 		// This PVC is required to exist to check its access modes.
-		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterPersistentVolumeClaimChange},
 	}
 }
 
+// isSchedulableAfterPodDeleted is invoked whenever a pod deleted,
+// It checks whether the deleted pod will conflict with volumes of other pods on the same node
+func (pl *VolumeRestrictions) isSchedulableAfterPodDeleted(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldPod, _, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPodDeleted: %w", err)
+	}
+
+	if oldPod.Namespace != pod.Namespace {
+		return framework.QueueSkip, nil
+	}
+
+	ctx := context.TODO()
+	newPodPvcs, err := pl.readWriteOncePodPVCsForPod(ctx, pod)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.V(2).Info("Failed to find pod's PersistentVolumeClaim", "pod", klog.KObj(pod), "err", err)
+			return framework.QueueSkip, nil
+		}
+		return framework.Queue, err
+	}
+
+	existingPodPvcs, err := pl.readWriteOncePodPVCsForPod(ctx, oldPod)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
What if the PVC is deleted before `isSchedulableAfterPodDeleted` is executed?

Reply:
Maybe we should return framework.Queue here? because it means that the current pod is likely to be schedulable.

Reply:
OK, I look deeply into readWriteOncePodPVCsForPod.
We can ignore a deleted PVC associated with this deleted Pod because:
- if a new Pod has this PVC, this Pod won't be schedulable until the PVC with the same name is recreated.
- if a new Pod doesn't have this PVC, this Pod is completely not related to that deleted PVC.

But, the difficulty is that what if the deleted Pod has two PVCs, and PVC1 is deleted, but PVC2 isn't deleted.
In this case, as I argued, PVC1 can be ignored anyway. But we still need to check PVC2.
We cannot simply return `Queue` because at this point we don't know PVC2 is related to a new Pod or not.
Could you change the implementation taking into this scenario into consideration?

Reply:
I've made changes in response to your comments, please see if they are as expected :)